### DISCUSSION TITLE
disable or silent access logs

### DISCUSSION THREAD

**Discussion by ThaiDangFr** [gollum/gollum#2071] [Q&A] [answered]
Hello, is there a way to disable or silent all the access logs ?
like this one for example
10.42.2.10 - - [10/Sep/2024:20:14:07 UTC] "GET /Home HTTP/1.1" 200 6306

**Comment by benjaminwil** [comment, marked as answer]
I only have a hacky solution for you! Your mileage may vary.

I cannot find a built-in way to disable or silence the access logs. Those access logs are generated by WEBrick, the application server that Gollum uses.

I didn't search the WEBrick documentation exhaustively, but I did find evidence that someone successfully silenced their access logs [by monkeypatching the WEBrick code from their application](https://stackoverflow.com/a/54195164).

So, I was able to silence the access log messages by adding the following code to my wiki's `config.rb` file (and ensuring that I started Gollum with the `--config` option):

```ruby
module WEBrickHTTPServerOverride
  RESPONSE_CODES_TO_IGNORE = [200, 302]

  def access_log(_config, _request, response)
    return super unless RESPONSE_CODES_TO_IGNORE.include?(response&.status)
  end

  WEBrick::HTTPServer.prepend self
end
```

It would be nice to stop using WEBrick and use a production-grade application server like Puma, which provides explicit configuration options to silence logging. But this would be more involved.

**Reply by ThaiDangFr** [reply]
Thanks @benjaminwil for your reply
I tried your solution but it doesn't seem to work for me
This is the logs :



```bash
[2024-09-11 16:31:07] INFO  WEBrick 1.8.1
[2024-09-11 16:31:07] INFO  ruby 3.1.6 (2024-05-29) [x86_64-linux-musl]
== Sinatra (v4.0.0) has taken the stage on 4567 for production with backup from WEBrick
[2024-09-11 16:31:07] INFO  WEBrick::HTTPServer#start: pid=1 port=4567
10.42.2.112 - - [11/Sep/2024:16:31:23 UTC] "GET / HTTP/1.1" 302 0
https://gollum.home/Home -> /
10.42.2.112 - - [11/Sep/2024:16:31:23 UTC] "GET /Home HTTP/1.1" 200 6306
https://gollum.home/Home -> /Home
10.42.2.112 - - [11/Sep/2024:16:31:34 UTC] "GET /bookmark.md HTTP/1.1" 200 12225
https://gollum.home/Home -> /bookmark.md
10.42.2.112 - - [11/Sep/2024:16:39:18 UTC] "GET / HTTP/1.1" 302 0
https://gollum.home/bookmark.md -> /
10.42.2.112 - - [11/Sep/2024:16:39:18 UTC] "GET /Home HTTP/1.1" 200 6306
https://gollum.home/bookmark.md -> /Home

```

I am using the docker.io/gollumwiki/gollum image.

Golum is launched like this :
```bash
/wiki $ ps -ef
UID          PID    PPID  C STIME TTY          TIME CMD
www-data       1       0  1 16:31 ?        00:00:02 /usr/local/bin/ruby -Eutf-8 /usr/local/bundle/bin/gollum --config=/config.rb

/wiki $ cat /config.rb 
module WEBrickHTTPServerOverride
  RESPONSE_CODES_TO_IGNORE = [200, 302]

  def access_log(_config, _request, response)
    return super if RESPONSE_CODES_TO_IGNORE.include?(response&.status)
  end

  WEBrick::HTTPServer.prepend self
end
```
Is there anything I miss ?

**Reply by benjaminwil** [reply]
Hi @ThaiDangFr.

Sorry for the confusion. I made a mistake in my code snippet: I meant `unless` rather than `if`. I have edited the code snippet in the original post to be correct now.

I just tested this in a Docker container and it seems to be working for me.

Here is the command I used, for reference:

```
docker run --rm --publish 4567:4567 --volume ~/my-wiki:/wiki gollumwiki/gollum --config /wiki/config.rb
```

**Reply by ThaiDangFr** [reply]
Thank you @benjaminwil, it works well now !
